BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsModule/DiagnosticsEventListener.cs

```diff
@@ -86,6 +86,34 @@ protected override void OnEventSourceCreated(EventSource eventSource)
         /// </summary>
         private static bool ShouldSubscribe(EventSource eventSource)
         {
+#if REDFIELD
+            if (eventSource.Name.StartsWith("Redfield-Microsoft-A", StringComparison.Ordinal))
+            {
+                switch (eventSource.Name)
+                {
+                    case "Redfield-Microsoft-ApplicationInsights-Core":
+                    case "Redfield-Microsoft-ApplicationInsights-WindowsServer-TelemetryChannel":
+
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-AppMapCorrelation-Dependency":
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-AppMapCorrelation-Web":
+
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-DependencyCollector":
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-EventCounterCollector":
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-PerformanceCollector":
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-PerformanceCollector-QuickPulse":
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-Web":
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-WindowsServer":
+                    case "Redfield-Microsoft-ApplicationInsights-WindowsServer-Core":
+                    case "Redfield-Microsoft-ApplicationInsights-Extensibility-EventSourceListener":
+                    case "Redfield-Microsoft-ApplicationInsights-AspNetCore":
+                    case "Redfield-Microsoft-ApplicationInsights-LoggerProvider":
+                    case "Microsoft-AspNet-Telemetry-Correlation":
```

Comment:
It is possible to hit this condition at all?

Reply:
Curious, what's the performance of this switch statement vs. a dictionary look up?

Reply:
Good catch. This needs to be fixed.

Reply:
Performance needs to be validated. Definitely, dictionary might be faster as we check 15 cases every time.

Reply:
I'm fuzzy on the exact details, I thought after a certain number of cases, a switch gets compiled as a hashmap?

Reply:
Performed some research with IL code and then hit an article which speaks about this scenario - https://stackoverflow.com/questions/11617091/in-a-switch-vs-dictionary-for-a-value-of-func-which-is-faster-and-why